sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java

```diff
@@ -425,13 +438,31 @@ private void readLongBatch(int rowId, int num, WritableColumnVector column) thro
     // This is where we implement support for the valid type conversions.
     if (column.dataType() == DataTypes.LongType ||
         DecimalType.is64BitDecimalType(column.dataType()) ||
-        originalType == OriginalType.TIMESTAMP_MICROS) {
-      defColumn.readLongs(
-        num, column, rowId, maxDefLevel, (VectorizedValuesReader) dataColumn);
-    } else if (originalType == OriginalType.TIMESTAMP_MILLIS) {
+        isTimestampWithUnit(logicalTypeAnnotation, MICROS)) {
+      if (needsTimezoneAdjustment()) {
+        for (int i = 0; i < num; i++) {
+          if (defColumn.readInteger() == maxDefLevel) {
+            long timestamp = dataColumn.readLong();
+            if (needsTimezoneAdjustment()) {
```

Comment:
Unnecessary if condition as`needsTimezoneAdjustment()` will be always true because in line 442 it is already tested.